### PR TITLE
Prepare for Release

### LINKED ISSUE #1144: System.Reflection.ReflectionTypeLoadException: The classes in the module cannot be loaded.
Just attempted to load my project in Visual Studio after developing solely in Xamarin Studio for the entire project duration.

I'm getting this error: System.Reflection.ReflectionTypeLoadException: The classes in the module cannot be loaded.

The error occurs during this call.
`_realm = Realm.GetInstance("salt.realm");`

The project type is Xamarin.Android with a PCL project and uses the latest version of Realm (0.81.0).
I've tried all the different linking options to no avail.

All my realm objects contain the attribute:
`[Preserve(AllMembers = true)]`

I'm also using Nuget v3 with the project.json file. Not sure if that makes a difference.

### LINKED ISSUE #1209: Release v1.0 as 1.0.3 or higher to avoid update errors
There is a NuGet problem with updates that bites occasionally in Xamarin Studio, due to the history of someone else first using that project identifier. Their highest version was 1.0.2.4.

It appears that the Xamarin Studio _update_ looks, quite reasonably, for the _highest version_ at the time of installing.

The following log is typical of an Android update **failing**.

Note that prior to selecting _update,_ Xamarin Studio had identified that I had 0.83.0 available 

Current workarounds I have applied:

1. Configure a local source and **disable** the global repo for the duration of the update running.
1. Remove Realm nuget and then add it, directly selecting the desired version. 

```
Attempting to gather dependency information for multiple packages with respect to project 'TestSyncAuth.Droid', targeting 'MonoAndroid,Version=v7.0'
Attempting to resolve dependencies for multiple packages.
Resolving actions install multiple packages
  GET https://www.nuget.org/api/v2/Packages(Id='Autofac',Version='3.5.2')
  OK https://www.nuget.org/api/v2/Packages(Id='Autofac',Version='3.5.2') 1177ms
For adding package 'Autofac.3.5.2' to project 'TestSyncAuth.Droid' that targets 'monoandroid70'.
For adding package 'Autofac.3.5.2' to project 'TestSyncAuth.Droid' that targets 'monoandroid70'.
Adding package 'Autofac.3.5.2' to folder '/Users/andydent/dev/Realm/realm-dotnet-private/TestSyncAuth/packages'
Added package 'Autofac.3.5.2' to folder '/Users/andydent/dev/Realm/realm-dotnet-private/TestSyncAuth/packages'
Added reference 'Autofac' to project 'TestSyncAuth.Droid'.
Added package 'Autofac.3.5.2' to 'packages.config'
Added file 'packages.config' to project 'TestSyncAuth.Droid'.
Successfully installed 'Autofac 3.5.2' to TestSyncAuth.Droid
  GET https://www.nuget.org/api/v2/Packages(Id='Realm',Version='1.0.2.4')
  OK https://www.nuget.org/api/v2/Packages(Id='Realm',Version='1.0.2.4') 1534ms
For adding package 'Realm.1.0.2.4' to project 'TestSyncAuth.Droid' that targets 'monoandroid70'.
Install failed. Rolling back...
Package 'Realm 1.0.2.4' does not exist in project 'TestSyncAuth.Droid'
Removed package 'Autofac 3.5.2' from 'packages.config'
Added file 'packages.config' to project 'TestSyncAuth.Droid'.
Removed reference 'Autofac' from project 'TestSyncAuth.Droid'.
Package 'Realm 1.0.2.4' does not exist in folder '/Users/andydent/dev/Realm/realm-dotnet-private/TestSyncAuth/packages'
Removing package 'Autofac 3.5.2' from folder '/Users/andydent/dev/Realm/realm-dotnet-private/TestSyncAuth/packages'
Removed folder '/Users/andydent/dev/Realm/realm-dotnet-private/TestSyncAuth/packages/Autofac.3.5.2/lib/portable-net4+sl5+netcore45+wpa81+wp8+MonoAndroid1+MonoTouch1'.
Removed folder '/Users/andydent/dev/Realm/realm-dotnet-private/TestSyncAuth/packages/Autofac.3.5.2/lib/net40'.
Removed folder '/Users/andydent/dev/Realm/realm-dotnet-private/TestSyncAuth/packages/Autofac.3.5.2/lib'.
Removed file 'Autofac.3.5.2.nupkg' from folder '/Users/andydent/dev/Realm/realm-dotnet-private/TestSyncAuth/packages/Autofac.3.5.2'.
Removed folder '/Users/andydent/dev/Realm/realm-dotnet-private/TestSyncAuth/packages/Autofac.3.5.2'.
Removed package 'Autofac 3.5.2' from folder '/Users/andydent/dev/Realm/realm-dotnet-private/TestSyncAuth/packages'
Could not install package 'Realm 1.0.2.4'. You are trying to install this package into a project that targets 'MonoAndroid,Version=v7.0', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.
```

### PR DESCRIPTION
**Pull request by nirinchev**
Fixes #1209
Fixes #1144

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+------------------
+# Out of Beta!
+After about a year and a half of hard work, we are proud to call this a 1.0 release. There is still work to do, but Realm Xamarin is now being used by thousands of developers and has proven reliable.
+
+## Sync
+Realm Xamarin now works with the Realm Mobile Platform. This means that you can write Xamarin apps that synchronize seamlessly with a Realm Object Server, allowing you to write complex apps with Xamarin that are offline-first and automatically synchronised literally by adding a few lines of configuration code.
+You can read about this in the [documentation](https://realm.io/docs/xamarin/latest/#sync)
+
+## Windows Desktop
+Realm Xamarin is no longer iOS and Android only. You can now use it to write .NET programs for Windows Desktop. Add the NuGet package to your regular .NET project and start using Realm.
```
Comment (nirinchev) [inline CHANGELOG.md]: Perhaps mention that Sync is not yet enabled, but we're working on it?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+Realm Xamarin now works with the Realm Mobile Platform. This means that you can write Xamarin apps that synchronize seamlessly with a Realm Object Server, allowing you to write complex apps with Xamarin that are offline-first and automatically synchronised literally by adding a few lines of configuration code.
+You can read about this in the [documentation](https://realm.io/docs/xamarin/latest/#sync)
+
+## Windows Desktop
+Realm Xamarin is no longer iOS and Android only. You can now use it to write .NET programs for Windows Desktop. Add the NuGet package to your regular .NET project and start using Realm. 
+
+### Bug fixes
+ - `[MapTo]` attribute is now considered in queries (#1219)
+ - Crashes in `RealmState.RemoveRealm` have been resolved (#1212)
+ - Update object by primary key on Android has been resolved (#1207)
```
Comment (nirinchev) [inline CHANGELOG.md]: The title of the issue is misleading. The actual bug is "Unsubscribing from `PropertyChanged` in a `PropertyChanged` callback should no longer lead to crashes"

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,23 @@
...
+## Sync
+Realm Xamarin now works with the Realm Mobile Platform. This means that you can write Xamarin apps that synchronize seamlessly with a Realm Object Server, allowing you to write complex apps with Xamarin that are offline-first and automatically synchronised literally by adding a few lines of configuration code.
+You can read about this in the [documentation](https://realm.io/docs/xamarin/latest/#sync)
+
+## Windows Desktop
+Realm Xamarin is no longer iOS and Android only. You can now use it to write .NET programs for Windows Desktop. Add the NuGet package to your regular .NET project and start using Realm. 
+
+### Bug fixes
+ - `[MapTo]` attribute is now considered in queries (#1219)
+ - Crashes in `RealmState.RemoveRealm` have been resolved (#1212)
```
Comment (nirinchev) [inline CHANGELOG.md]: Again - the issue title is a bit misleading - It's more like "Letting a Realm be finalized instead of disposing it will no longer lead to crashes"

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,23 @@
...
+ - Update object by primary key on Android has been resolved (#1207)
+ - `WriteAsync` now advances the read transaction (#1192)
+ - Queries on backlinks no longer fail (#1177)
```
Comment (nirinchev) [inline CHANGELOG.md]: They didn't fail, more like - didn't produce correct results :P I suggest rewording it to "Queries on backlink properties should no longer produce unexpected results"

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,23 @@
+1.0.3 (2017-02-14)
+------------------
+# Out of Beta!
```
Comment (hrach) [inline CHANGELOG.md]: I would put here some exaplanation why 1.0.3 and not 1.0.0. (I know the reason)

Comment (kristiandupont) [inline CHANGELOG.md]: :-) I thought of it but I think it would mostly be confusing to people.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,26 @@
...
+Realm Xamarin is no longer iOS and Android only. You can now use it to write .NET programs for Windows Desktop. Add the NuGet package to your regular .NET project and start using Realm. Sync does not yet work for Windows, but we are working on it and you can expect support soon.
+
+### Breaking Changes
+ - `IRealmCollection<T>.ObjectSchema` is deprecated and replaced with `ISchemaSource.ObjectSchema`. (#1216) 
+
+### Bug fixes
+ - `[MapTo]` attribute is now considered in queries (#1219)
+ - Letting a Realm be finalized instead of disposing it will no longer lead to crashes (#1212)
+ - Unsubscribing from PropertyChanged in a PropertyChanged callback should no longer lead to crashes (#1207)
+ - `WriteAsync` now advances the read transaction (#1192)
```
Comment (AndyDentFree) [inline CHANGELOG.md]: This sounds a bit obscure, maybe add _ensuring other threads see the effects of the Write_ ?
